### PR TITLE
feat(java-client): java client add class host_port for fqdn compile

### PR DESCRIPTION
**Pull request by Samunroyu**
Make java client compile success with class host_port.

### CODE REVIEW

File: java-client/src/main/java/org/apache/pegasus/base/host_port.java [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+  public boolean equals(host_port that) {
+    if (that == null) return false;
+    if (this == that) return true;
+
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    int hashCode = 1;
```
Comment (acelyc111) [inline java-client/src/main/java/org/apache/pegasus/base/host_port.java:203]: You need to implement a real hash function.

File: java-client/src/main/java/org/apache/pegasus/base/host_port.java [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    return hashCode;
+  }
+
+  @Override
+  public int compareTo(host_port other) {
+    if (!getClass().equals(other.getClass())) {
+      return getClass().getName().compareTo(other.getClass().getName());
+    }
+
+    int lastComparison = 0;
```
Comment (acelyc111) [inline java-client/src/main/java/org/apache/pegasus/base/host_port.java:214]: You need to compare the actual content.

File: java-client/src/main/java/org/apache/pegasus/base/host_port.java [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+  public void read(org.apache.thrift.protocol.TProtocol iprot) throws org.apache.thrift.TException {
+    host = iprot.readString();
+    port = iprot.readI16();
+    hyType = iprot.readByte();
+
+    validate();
+  }
+
+  public void write(org.apache.thrift.protocol.TProtocol oprot)
+      throws org.apache.thrift.TException {
```
Comment (acelyc111) [inline java-client/src/main/java/org/apache/pegasus/base/host_port.java:232]: validate() at first.

File: java-client/src/main/java/org/apache/pegasus/base/host_port.java [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+    oprot.writeString(host);
+    oprot.writeI16(port);
+    byte byte_hyType = (byte) (hyType);
+    oprot.writeByte(byte_hyType);
+  }
+
+  @Override
+  public java.lang.String toString() {
+    java.lang.StringBuilder sb = new java.lang.StringBuilder("host_port(");
+    boolean first = true;
```
Comment (acelyc111) [inline java-client/src/main/java/org/apache/pegasus/base/host_port.java:243]: You need to use the actual content.

File: java-client/src/main/java/org/apache/pegasus/base/host_port.java [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+    sb.append(")");
+    return sb.toString();
+  }
+
+  public void validate() throws org.apache.thrift.TException {
+    // check for required fields
+    // check for sub-struct validity
+  }
+
+  private void writeObject(java.io.ObjectOutputStream out) throws java.io.IOException {
```
Comment (acelyc111) [inline java-client/src/main/java/org/apache/pegasus/base/host_port.java:254]: Are the following functions necessary?

File: java-client/src/main/java/org/apache/pegasus/base/host_port.java [hunk 6]
```diff
@@ -0,0 +1,348 @@
+/*
```
Comment (acelyc111) [inline java-client/src/main/java/org/apache/pegasus/base/host_port.java:1]: Add tests to check the basic functions, at least read() and write().

File: java-client/src/main/java/org/apache/pegasus/base/host_port.java [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+  private static final org.apache.thrift.scheme.SchemeFactory STANDARD_SCHEME_FACTORY =
+      new host_portStandardSchemeFactory();
+  private static final org.apache.thrift.scheme.SchemeFactory TUPLE_SCHEME_FACTORY =
+      new host_portTupleSchemeFactory();
+
+  public String host;
+
+  public short port;
+
+  public int hyType;
```
[the remark this answers is not in the document]
Comment (Samunroyu) [reply java-client/src/main/java/org/apache/pegasus/base/host_port.java:49]: It's typo.
But is the same meaning with hpType in go-client/idl/base/host_port.go. The hpType is mean hostPortType.

File: java-client/src/test/java/org/apache/pegasus/base/TestHostPort.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    TMemoryBuffer buffer = new TMemoryBuffer(500);
+    TBinaryProtocol protocol = new TBinaryProtocol(buffer);
+    hostPort.write(protocol);
+
+    host_port decodeHostPort = new host_port();
+    decodeHostPort.read(protocol);
+
+    assertEquals(hostPort.getHost(), decodeHostPort.getHost());
+    System.out.println(hostPort.getPort());
```
Comment (acelyc111) [inline java-client/src/test/java/org/apache/pegasus/base/TestHostPort.java:53]: Remove the debuging code.

**Review by acelyc111 — APPROVED** [review]

**Review by foreverneverer — APPROVED** [review]

**Pull request merged**